lua/acf/shared/armor/ceramic.lua

```diff
@@ -21,46 +24,46 @@ Material.ArmorMul	= 1.8
 Material.NormMult	= 1.5
 
 if SERVER then
-	function Material.ArmorResolution( _, armor, losArmor, losArmorHealth, maxPenetration, FrArea, caliber, damageMult, Type)
+	function Material.ArmorResolution( Entity, armor, losArmor, losArmorHealth, maxPenetration, FrArea, _, damageMult, Type)
 
 		local HitRes = {}
 
 		local curve		= Material.curve
 		local effectiveness = Material.effectiveness
 		local resiliance	= Material.resiliance
 
+		local ductilityvalue = (Entity.ACF.Ductility or 0) * 1.25 --The ductility value of the armor. Outputs 1 to -1 depending on max ductility
+		local ductilitymult    = 2 / (2 - ductilityvalue * 1.5) -- Direct damage multiplier based on ductility.
+
 		armor = armor ^ curve
 		losArmor = losArmor ^ curve
 
-		local slopeDmg = ( losArmor / armor ) --Angled ceramic takes more damage. Fully angled ceramic takes up to 7x the damage
+		local dmul = ( losArmor / armor ) --Angled ceramic takes more damage. Fully angled ceramic takes up to 7x the damage
 
 		if Type == "HE" or Type == "HESH" then
-			slopeDmg = slopeDmg * 5
+			dmul = dmul * 15
 		end
 
-		local dmul = slopeDmg
-
-		-- Breach probability
-		local breachProb = math.Clamp((caliber / armor / effectiveness - 1.3) / (7 - 1.3), 0, 1)
+		--Removed Breaches from Ceramic. This was hard to balance, complicated calculations, and shouldn't really happen.
 
 		-- Penetration probability
+		--The larger the number on the inside, the lower the penetration probability
+		--Penetration/Armor ratios below 1 lead to ludicrously large numbers, making penetration nearly impossible.
+		--Ratios of 1-2 lead to extremely small numbers around 1, causing penetration chances of ~40%-99%
+		--Ratios larger than 2 lead to ludicrously small numbers, making penetration almost guarenteed.
 		local penProb = (math.Clamp(1 / (1 + math.exp(-43.9445 * (maxPenetration / losArmor / effectiveness - 1))), 0.0015, 0.9985) - 0.0015) / 0.997;
 
-		-- Breach chance roll
-		if breachProb > math.random() and maxPenetration > armor then
-
-			HitRes.Damage	= FrArea / resiliance * damageMult * dmul	-- Inflicted Damage
-			HitRes.Overkill = maxPenetration - armor												-- Remaining penetration
-			HitRes.Loss	= armor / maxPenetration												-- Energy loss in percents
-
-			return HitRes
-
 		-- Penetration chance roll
-		elseif penProb > math.random() then
+		if penProb > math.random() then
 
 			local Penetration = math.min( maxPenetration, losArmor * effectiveness )
 
-			HitRes.Damage	= ( Penetration / losArmorHealth / effectiveness ) ^ 2 * FrArea / resiliance * damageMult * dmul
+			if maxPenetration > losArmor * effectiveness then
+				dmul = dmul * 20 --Damage multiplier for ceramic when it gets penned. Is this enough shatter?
```

Comment:
hardcoded multiplier. It'd be better if you'd make a variable like Material.DamageMul (or any other name) and adjust it in the script.